pkg/compactor/planned_jobs_http.go

```diff
@@ -0,0 +1,178 @@
+package compactor
+
+import (
+	_ "embed"
+	"fmt"
+	"html/template"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/oklog/ulid"
+	"github.com/prometheus/prometheus/model/timestamp"
+
+	"github.com/grafana/mimir/pkg/storage/bucket"
+	"github.com/grafana/mimir/pkg/storage/tsdb"
+	"github.com/grafana/mimir/pkg/storage/tsdb/bucketindex"
+	"github.com/grafana/mimir/pkg/util"
+)
+
+//go:embed compactor_tenants.gohtml
+var tenantsPageHTML string
+var tenantsTemplate = template.Must(template.New("webpage").Parse(tenantsPageHTML))
+
+type tenantsPageContents struct {
+	Now     time.Time `json:"now"`
+	Tenants []string  `json:"tenants,omitempty"`
+}
+
+func (c *MultitenantCompactor) TenantsHandler(w http.ResponseWriter, req *http.Request) {
+	tenants, err := tsdb.ListUsers(req.Context(), c.bucketClient)
+	if err != nil {
+		util.WriteTextResponse(w, fmt.Sprintf("Can't read tenants: %s", err))
+		return
+	}
+
+	util.RenderHTTPResponse(w, tenantsPageContents{
+		Now:     time.Now(),
+		Tenants: tenants,
+	}, tenantsTemplate, req)
+}
+
+//go:embed planned_jobs.gohtml
+var plannerJobsHTML string
+var plannerJobsTemplateFuncs = template.FuncMap{"add": func(x, y int) int { return x + y }}
+var plannerJobsTemplate = template.Must(template.New("webpage").Funcs(plannerJobsTemplateFuncs).Parse(plannerJobsHTML))
+
+type plannerJobsContent struct {
+	Now                string `json:"now"`
```

Comment:
Why is this a string, and a `time.Time` in `tenantsPageContents`? And vice versa.

Reply:
Having loaded the other page, maybe `Now` should be a formatted string there also?

Reply:
Having checked offline, not string formatting the time on the other page seems intentional.